lib/Runtime/Library/JsBuiltInEngineInterfaceExtensionObject.cpp

```diff
@@ -0,0 +1,425 @@
+                JavascriptOperators::GetPropertyId(alias, scriptContext);
+            library->AddMember(prototype, aliasFunctionIdentifier, scriptFunction);
+        }
+
+        if (functionIdentifier == PropertyIds::values)
+        {
+            library->arrayPrototypeDefaultValuesFunction = func;
+        }
+
+        //Don't need to return anything
+        return library->GetUndefined();
+    }
+
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_GetLength(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+        Assert(callInfo.Count == 2);
+
+        int64 length;
+        Var iterable = args.Values[1];
+
+        TypedArrayBase *typedArrayBase = nullptr;
+        Assert(!JavascriptArray::Is(iterable));
+
+        if (TypedArrayBase::Is(iterable))
+        {
+            typedArrayBase = TypedArrayBase::FromVar(iterable);
+            if (typedArrayBase->IsDetachedBuffer())
+            {
+                JavascriptError::ThrowTypeError(scriptContext, JSERR_DetachedTypedArray);
+            }
+
+            length = typedArrayBase->GetLength();
+        }
+        else
+        {
+            length = JavascriptConversion::ToLength(JavascriptOperators::OP_GetLength(iterable, scriptContext), scriptContext);
+        }
+
+        return JavascriptNumber::ToVar(length, scriptContext);
+    }
+
+    /*
+    * First parameter is the object onto which prototype should be set; second is the value
+    */
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_SetPrototype(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+        UNREFERENCED_PARAMETER(scriptContext);
+
+        Assert(callInfo.Count == 3 && DynamicObject::Is(args.Values[1]) && RecyclableObject::Is(args.Values[2]));
+
+        DynamicObject* obj = DynamicObject::FromVar(args.Values[1]);
+        RecyclableObject* value = RecyclableObject::FromVar(args.Values[2]);
+
+        obj->SetPrototype(value);
+
+        return obj;
+    }
+
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_GetIteratorPrototype(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+
+        Assert(callInfo.Count == 1);
+
+        return scriptContext->GetLibrary()->iteratorPrototype;
+    }
+
+    Var JsBuiltInEngineInterfaceExtensionObject::EntryJsBuiltIn_Internal_InitInternalProperties(RecyclableObject *function, CallInfo callInfo, ...)
+    {
+        EngineInterfaceObject_CommonFunctionProlog(function, callInfo);
+
+        DynamicObject* obj = DynamicObject::FromVar(args.Values[1]);
+        unsigned propCount = TaggedInt::ToUInt32(args.Values[2]);
+        
+        Assert(callInfo.Count == 3 + propCount);
```

Comment:
you don't need this, right? this function can be used for other functions in the future.

Reply:
Assert will stay the same for all new functions as well. 3 here is this + object on which we want to set property attributes + count of properties